server/public/model/channel_mentions.go

```diff
@@ -26,3 +28,22 @@ func ChannelMentions(message string) []string {
 
 	return names
 }
+
+func ChannelMentionsAcrossTeams(message string) map[string][]string {
```

Comment:
Let's add a test for this.

Also, can we add some docs to explain the format of the `map[string][]string`?